### DISCUSSION TITLE
Import external html

### DISCUSSION THREAD

**Discussion by engineer-myoa** [ArchiveBox/ArchiveBox#862] [Ideas]
### Abstract
- Suggest feature for import html via external source to ArchiveBox

### Description
Archivebox is most powerful archiving tool.
But there are several weaknesses.
I always been regretted when using archiving, It can not archive which is hidden resources.
Like these, if you are using a VPN or require to authentication or on an intranet.
So I suggest how about import singular file from external sources. I'm using on SingleFile.(Ref. a)
This is one of my favorite methods, when I need to archive hidden resources.
But, Actually, It made really really hell of management of files.

In conclusion, How about import singular html file to archivebox?
Even if it does not have original uri, and other features, I think that it is really good.
Hopefully. :)

Regards.

### References
- a. https://github.com/gildas-lormeau/SingleFile

**Comment by pirate** [comment]
Why not create a Snapshot in ArchiveBox with the inaccessible URL, let it fail, then replace `archive/<timestamp>/singlefile.html` on the filesystem with the file you want to import. ArchiveBox will treat it as its own archive and show it correctly in the UI.

**Reply by anirudhmurali** [reply]
This works, but importing existing Singlefiles would be helpful.

**Reply by ccxuy** [reply]
it would be perfect if there is a chrome extension could do this all together with one click
